torch/distributed/_tensor/experimental/attention.py

```diff
@@ -231,6 +232,11 @@ def _templated_ring_attention(
 
     chunks = []
     logsumexps = []
+    # Without making key and value contiguous(), the lose curve is bad.
+    # TODO(fegin): figure out why this is a requirement since SDPA does not have
+    # this requirement.
```

Comment:
@drisspg should we be checking this in the sdpa operators?

Reply:
We do have many checks within the operators both at the runtime dispatcher, however I see that this coe is not calling the 'can use functions'
https://github.com/pytorch/pytorch/blob/9554a9af8788c57e1c5222c39076a5afcf0998ae/torch/nested/_internal/sdpa.py#L278
https://github.com/pytorch/pytorch/blob/9554a9af8788c57e1c5222c39076a5afcf0998ae/torch/nested/_internal/sdpa.py#L297

As well within the kernel we have last line of dense checks:
https://github.com/pytorch/pytorch/blob/main/aten/src/ATen/native/transformers/cuda/flash_attn/flash_api.cpp#L369-L393

I really dont know why the contigous calls are needed unless there is some other invariant not being satisfied